### PR TITLE
Fix process loading stuck

### PR DESCRIPTION
**Pull request by antonrohr**
This fixes http://b/182996343 by starting `state_machine_` before registering the `OnProcessListUpdate` callback with `process_manager_` (Only the "Fix ProfilingTargetDialog being stuck" is relevant. The other commits are just refactoring/renaming). 

I am pretty sure this solved the problem, but it would be appreciated if someone could test this PR (@karupayun, the problem seems to occur often on your machine, maybe you can test it?)

This is a PR for main and I dont think it needs to be on release, since I dont think this is a regression, but I am happy to be convinced otherwise

### CODE REVIEW

**Comment by beckerhe** [comment]
Haven't looked at it yet. But I wanted to say I agree. This doesn't need to be in the release.

File: src/OrbitQt/ProfilingTargetDialog.cpp [hunk 1]
```diff
@@ -144,7 +144,8 @@ ProfilingTargetDialog::ProfilingTargetDialog(
   if (target_configuration_opt.has_value()) {
     TargetConfiguration config = std::move(target_configuration_opt.value());
     target_configuration_opt = std::nullopt;
-    SetStateMachineInitialStateFromTarget(std::move(config));
+    std::visit([this](auto target) { SetTargetAndStateMachineInitialState(std::move(target)); },
```
Comment (beckerhe) [inline src/OrbitQt/ProfilingTargetDialog.cpp:147]: Nit: `auto&& target` - Will save a move. But is also super-nit!

**Review by beckerhe — APPROVED** [review]
LGTM

Thanks a lot for splitting this up so fine-grained. Made the review a breeze!

I can also test this later or tomorrow morning.

**Comment by beckerhe** [comment]
Trying this now.

File: src/OrbitQt/ProfilingTargetDialog.cpp (hunk 1 above)
Comment (antonrohr) [reply src/OrbitQt/ProfilingTargetDialog.cpp]: I added `&&` and removed one `std::move`.

**Comment by beckerhe** [comment]
> Trying this now.

Seems to work. I can't reproduce the problem anymore! Thanks!

**Review by karupayun — APPROVED** [review]
LGTM! Thanks for solving this!

Before these 4 commits, I made 5 tries before reproduced it (20%), after I tried 12 times and couldn't reproduce it.

No comments in the code.

**Comment by antonrohr** [comment]
Okay, this sounds good!
There is another issue with switching between local and stadia that @reichlfl found (see http://b/182996343). I will address this in different PR.

**Pull request merged**